src/drivers/stm32/drv_io_timer.c

```diff
@@ -457,7 +457,7 @@ static int timer_set_rate(unsigned timer, unsigned rate)
 {
 
 	/* configure the timer to update at the desired rate */
-	rARR(timer) = BOARD_PWM_FREQ / rate;
+	rARR(timer) = (BOARD_PWM_FREQ / rate) - 1;
```

Comment:
@davids5 Could you please review?

Reply:
@davids5: Take note that "rate" is an unsigned parameter, so the division is integer division. I don't know whether we should typecast it to perform floating point division and then typecast that result to an integer before subtracting the 1.